pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -291,6 +291,21 @@ public static class QueryOptionKey {
 
         public static final String ORDER_BY_ALGORITHM = "orderByAlgorithm";
 
+        // Number of rows limit to select in leaf stage.
+        // It has to be greater than 0.
+        // If it is invalid number format, or the number is smaller than or equal to zero, we use
+        // DEFAULT_LEAF_NODE_LIMIT.
+        public static final String MS_LEAF_LIMIT = "multiStageLeafLimit";
+
+        // Override numGroupLimit used by v1 engine.
+        public static final String NUM_GROUP_LIMIT = "numGroupLimit";
+
+        // Override maxInitialResultCap used by v1 engine.
+        public static final String MAX_INITIAL_RESULT_HOLDER_CAPACITY = "maxInitialResultCap";
+
+        // Override groupByTrimThreshold used by v1 engine.
+        public static final String GROUP_BY_TRIM_THRESHOLD = "groupByTrimThreshold";
```

Comment:
@Jackie-Jiang ^ any idea whether we want to allow modify on all 3 or should we derive them from a basic setting?

Reply:
@snleee any thoughts on this^

Reply:
We can make all of them configurable from the query option to be flexible

Reply:
I think we have typically played with `numGroupsLimit` vast majority of the times and for some of our production use cases have set it higher than default 100K to get more accurate results for high cardinality GROUP BY. 

I don't remember configuring the `resultHolderCapacity` for any use case.

We have configured `trimThreshold` in one case for performance reasons

Reply:
Also IIRC, the `trimThreshold` option when it was introduced earlier was specifically applied at server level execution during combine in `ConcurrentIndexedTable`. Later it was pushed down into the segment level as well afaik. So, will be good to double check the expectations / implications of setting this in the context of multi-stage depending on desired behavior.

Reply:
sounds good. thanks for sharing your opinions